community/modules/compute/schedmd-slurm-gcp-v5-partition/main.tf

```diff
@@ -69,6 +69,16 @@ locals {
   # Also, slurm imposed a lot of restrictions to this name, so we format it to an acceptable string
   tmp_cluster_name   = substr(replace(lower(var.deployment_name), "/^[^a-z]*|[^a-z0-9]/", ""), 0, 8)
   slurm_cluster_name = var.slurm_cluster_name != null ? var.slurm_cluster_name : local.tmp_cluster_name
+
+  uses_zone_policies = length(var.zone_policy_allow) + length(var.zone_policy_deny) > 0
+  excluded_zones     = var.zone == null ? [] : [for z in data.google_compute_zones.available.names : z if z != var.zone]
+  zone_policy_deny   = local.uses_zone_policies ? var.zone_policy_deny : local.excluded_zones
+  zone_policy_allow  = local.uses_zone_policies || var.zone == null ? var.zone_policy_allow : [var.zone]
+}
+
+data "google_compute_zones" "available" {
+  project = var.project_id
+  region  = var.region
```

Comment:
Strictly speaking you are making the assumption that `var.zone` is in `var.region` which we do not know to be the case. I think you are making the most reasonable choice here but we should be aware that this solution is fragile because I believe it would have the impact of rejecting all zones if both `var.region` and `var.zone` are set without setting explicit allow/deny lists.

This would be a good use case for Terraform 1.2 pre/post conditions which could allow a test of whether `var.zone` was found in the zones data source.

Reply:
These 3 comments are all right around nit level. I am going to collect them in a task and approve PR.